src/pages/index.js

```diff
@@ -1,118 +1,174 @@
 import UserInfo from "../components/UserInfo.js";
-
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
+
+const api = new Api({
+    baseUrl: 'https://around-api.en.tripleten-services.com/v1',
+    headers: {
+        authorization: "77f6fa17-673a-40ba-89df-6b222f24f8b5",
+        'Content-Type': 'application/json'
+    }
+});
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  ".cards__list"
-);
+const userInfo = new UserInfo(profileTitle, profileDescription, document.querySelector('.profile__image'));
 
-section.renderItems();
+// Inicializar perfil
+api.getUserInfo()
+    .then(userData => {
+        userInfo.setUserInfo(userData);
+        userInfo.setAvatar(userData.avatar); 
+    })
+    .catch(err => console.error('Error loading user info:', err));
 
+// Crear cards
+const section = new Section(
+    {
+        items: [],
+        renderer: (cardData) => {
+            const cardElement = createCard(cardData);
+            if (cardElement) {
+                section.addItem(cardElement);
+            }
+        }
+    },
+    ".cards__list"
+);
 
-function createCard(data) {
-  const cardElement = getCardElement(data);
-  section.addItem(cardElement);
+api.getInitialCards()
+    .then(cardsData => {
+        cardsData.forEach(cardData => {
+            const cardElement = createCard(cardData);
+            if (cardElement) {
+                section.addItem(cardElement);
+            }
+        });
+    })
+    .catch(err => console.error('Error loading cards from API:', err));
+
+function createCard(cardData) {
+    const card = new Card({
+        cardData,
+        cardSelector: "#card-template",
+        handleImageClick,
+        handleDeleteClick: (cardId, cardElement) => handleDeleteCard(cardId, cardElement),
+        userId: userInfo.getUserInfo()._id,
+        handleLikeButton: (cardId, isLiked) => handleLikeClick(cardId, isLiked)
+    });
+    return card.getView();
 }
 
-function getCardElement(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
-}
+const imageModal = new PopupWithImage("#modal__preview-card");
+imageModal.setEventListeners();
 
 function handleImageClick(cardData) {
-  imageModal.open(cardData);
+    imageModal.open(cardData);
 }
 
-function handleAddCardFormSubmit(inputData) {
-  const cardData = {
-    name: inputData.title,
-    link: inputData.url,
-  };
-
-  createCard(cardData);
-  newCardModal.close();
-  addCardForm.reset();
-  addCardFormValidator.disableSubmitButton();
-}
-
-function handleProfileFormSubmit(userData) {
-  const title = userData.title;
-  const description = userData.description;
-  userInfo.setUserInfo(title, description);
-  profileModal.close();
-}
-
-const userInfo = new UserInfo(profileTitle, profileDescription);
+const profileModal = new PopupWithForm("#profile-edit-modal", (data) => {
+    profileModal.renderLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```